### ISSUE TITLE
Parsing CODEOWNERS sections

### ISSUE THREAD

**Issue by jirislav** [sbdchd/codeowners#20]
Hi,

we have started using your library and we have GitLab Premium, which supports sections, see:
- https://docs.gitlab.com/ee/user/project/code_owners.html#code-owners-sections

But this library seems to silently fail, when attempting to parse it.

Here is an `CODEOWNERS` example:
```codeowners
[Documentation maintainers]
README.md    @our.group

[Configuration maintainers]
/conf        @another.group
```

And when we use it in your library:
```python
 $ python
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210220] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from codeowners import CodeOwners
>>> owners = CodeOwners("CODEOWNERS")
>>> owners
<codeowners.CodeOwners object at 0x7f9058850070>
>>> owners.
owners.of(    owners.paths  
>>> owners.paths
[(re.compile('(?:\\A|/)CODEOWNERS(?:\\Z|/)'), [])]
>>> owners.of("README.md")
[]
>>> 
```
- you can see empty `paths` and also that `CodeOwners.of` doesn't return anything
- but the class is constructed without a fail

We use the latest version:
```bash
 $ pip freeze | grep codeowners 
codeowners==0.2.1
```

**Comment by sbdchd** [comment]
I wonder if this is gitlab specific or if other services use the same section syntax

Parsing the sections looks a little more involved than the current syntax supported

**Comment by jirislav** [comment]
> I wonder if this is gitlab specific or if other services use the same section syntax
> 
> Parsing the sections looks a little more involved than the current syntax supported

I think it's GitLab specific :/ . 

Since we only care for the library to not fail, how about just implementing some kind of ignorance?

e.g.:
```python
re.sub(r'^\^?\[[^\]]+\]', '', content)
```

**Comment by jirislav** [comment]
I've created a PR, there was no need to add a regex as it was easy to add to the same logic as comments have.

**Comment by sbdchd** [comment]
Sorry for reverting your PR, CI should now run for forks -- doesn't seem to do that by default

There were a couple tests that failed:

https://app.circleci.com/pipelines/github/sbdchd/codeowners/56/workflows/09ca8fba-9eaa-46fd-b0b7-10a28dce4668/jobs/120

specifically these two test cases:

https://github.com/sbdchd/codeowners/blob/d47302647bab6b049149f01c0e0f956f5122b3c5/codeowners/test_codeowners.py#L276-L284

**Comment by sbdchd** [comment]
Should be fixed in version 0.3.0

**Comment by jirislav** [comment]
I'm so sorry, I should've run the tests locally 🙈 ... but that's the price of GitHub Web editor and being used to automated CI checks 😅  .. thanks for the fix btw! and also for enabling CI in forks 👍🏿 great job!
